alert_group_settings.go

```diff
@@ -0,0 +1,119 @@
+package mackerel
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+)
+
+// AlertGroupSetting represents a Mackerel alert group setting.
+// ref. https://mackerel.io/api-docs/entry/alert-group-settings
+type AlertGroupSetting struct {
+	ID                   string   `json:"id,omitempty"`
+	Name                 string   `json:"name"`
+	Memo                 string   `json:"memo,omitempty"`
+	ServiceScopes        []string `json:"serviceScopes,omitempty"`
+	RoleScopes           []string `json:"roleScopes,omitempty"`
+	MonitorScopes        []string `json:"monitorScopes,omitempty"`
+	NotificationInterval uint64   `json:"notificationInterval,omitempty"`
+}
+
+// FindAlertGroupSettings finds alert group settings
+func (c *Client) FindAlertGroupSettings() ([]*AlertGroupSetting, error) {
+	req, err := http.NewRequest("GET", c.urlFor("/api/v0/alert-group-settings").String(), nil)
+	if err != nil {
+		return nil, err
+	}
+	resp, err := c.Request(req)
+	defer closeResponse(resp)
+	if err != nil {
+		return nil, err
+	}
+
+	var data struct {
+		AlertGroupSettings []*AlertGroupSetting `json:"alertGroupSettings"`
```

Comment:
According to the API docs as follows:
> ```
> {
>   "alert_group_settings": [<alertGroupSetting>, <alertGroupSetting>, ...]
> }
> ```

but it's actually as follows:
```
{
  "alertGroupSettings": [<alertGroupSetting>, <alertGroupSetting>, ...]
}
```

Reply:
Right. We'll fix the docs soon.

Reply:
Deployed fix for [the API doc](https://mackerel.io/api-docs/entry/alert-group-settings#list).
ref: https://github.com/mackerelio/documents/commit/356018e3c99fe03b8e58218ccbe670dace8f526e